### ISSUE TITLE
How to use this to send error response?

### ISSUE THREAD

**Issue by mcalthrop** [SeyZ/jsonapi-serializer#121]
Seems like a really silly question, but... how do I use this package to send an error response?

**Comment by ozanmuyes** [comment]
I don't think that's a silly question, I think errors beyond this library's responsibility. In order to implement the errors functionality, the HTTP errors should be defined upfront, and they will be application specific since neither NodeJS nor this library defines.
But, it would be good to have a default error handler and write `return next(new JSONAPINotFoundError('and then the "details"', 'also the "code" if app defines one', {links: 'as well'}, {meta: 'too'}));`
I would happily write the necessary codes if it's OK with the @SeyZ to merge the PR.

**Comment by SeyZ** [comment]
@ozanmuyes I totally agree, it would be awesome to have your PR! Thank you!

Just to be sure about the spec and the syntax:

Spec: http://jsonapi.org/format/#errors

Syntax:

new JSONAPINotFoundError({
  id: 'xxx',
  status: '400',
  title: '...',
  // etc. Check all opts there: http://jsonapi.org/format/#error-objects
});

Thoughts?

**Comment by ozanmuyes** [comment]
@SeyZ The syntax can be the same as yours, explicitly stated in an object which then passed to the desired error function (JSONAPINotFoundError in this manner); and, I also thought it will be handy to have implicit error functions, essentially the same error function which determines the behaviour considering the type of the first argument. It will be clearer as soon as I open the PR.

Also the error function/class name above was just an improvisation, I think it is too long and verbose. Maybe it would be better to call it as `new NotFound(...)`. What are your thoughts?

TL;DR Alright, the I agree with the syntax, by that compliancy fulfilled. Also error instantiation can be explicit and implicit, but shorter error names would be nicer. Thoughts?

**Comment by SeyZ** [comment]
It's now available 👍
